### ISSUE TITLE
RC test: Gravity Force Lab 2.2.0-rc.1

### ISSUE THREAD

**Issue by zepumph** [phetsims/QA#471]
@emily-phet @terracoda @jessegreenberg @arouinfar @kathy-phet, gravity-force-lab/2.2.0-rc.1 is ready for RC testing. 
Document issues in https://github.com/phetsims/gravity-force-lab/issues and link to this issue.

This simulation is outfitted with:
* All PhET-iO featured, including a designed api
* Sound
* Interactive Descriptions (alternative input and auditory descriptions)
* Gesture Accessibility with support for mobile iOS 

Assigning to @ariel-phet for prioritization.

<details>
<summary><b>General RC Test</b></summary>

<h3>What to Test</h3>

- Click every single button.
- Test all possible forms of input.
  - Test all mouse/trackpad inputs.
  - Test all touchscreen inputs.
- If there is sound, make sure it works.
- Make sure you can't lose anything.
- Play with the sim normally.
- Try to break the sim.
- Test all query parameters on all platforms. (See [QA Book](https://github.com/phetsims/QA/blob/master/doc/qa-book.md) for a list of query parameters.)
- Download HTML on Chrome and iOS.
- Make sure the iFrame version of the simulation is working as intended on all platforms.
- Make sure the XHTML version of the simulation is working as intended on all platforms.
- Complete the test matrix.
- Don't forget to make sure the sim works with Legends of Learning.
- Check [this](https://docs.google.com/spreadsheets/d/1umIAmhn89WN1nzcHKhYJcv-n3Oj6ps1wITc-CjWYytE/edit#gid=0) LoL spreadsheet and notify AR or AM if it not there.
- If this is rc.1 please do a memory test.
- When making an issue, check to see if it was in a previously published version
- If there is a console available, check for errors and include them in the Problem Description.
- As an RC begins and ends, check the sim repo. If there is a maintenance issue, check it and notify developers if there is a problem.

<h3>Link(s)</h3>

- **[Simulation](https://phet-dev.colorado.edu/html/gravity-force-lab/2.2.0-rc.1/phet/gravity-force-lab_all_phet.html)**
- **[iFrame](https://phet-dev.colorado.edu/html/gravity-force-lab/2.2.0-rc.1/phet/gravity-force-lab_all_iframe_phet.html)**
- **[XHTML](https://phet-dev.colorado.edu/html/gravity-force-lab/2.2.0-rc.1/phet/xhtml/gravity-force-lab_all.xhtml)**
- **[Test Matrix](https://docs.google.com/spreadsheets/d/1dPp2zsj4WW577vQNgCnx9d3cPNxh_JrJj_EQ8CaNqOc/edit#gid=1975614346)**
- **[Legends of Learning Harness](https://developers.legendsoflearning.com/public-harness/index.html?startPayload=%7B%22languageCode%22%3A%22en%22%7D)**

<hr>

</details>

<details>
<summary><b>PhET-iO RC Test</b></summary>

<h3>What to Test</h3>

- Make sure that public files do not have password protection. Use a private browser for this.
- Make sure that private files do have password protection. Use a private browser for this.
- Make sure standalone sim is working properly.
- Make sure the wrapper index is working properly.
- Make sure each wrapper is working properly.
- Launch the simulation in Studio with ?stringTest=xss and make sure the sim doesn't navigate to youtube
- For newer PhET-iO wrapper indices, save the "basic example of a functional wrapper" as a .html file and open it. Make sure the simulation loads without crashing or throwing errors.
- Load the login wrapper just to make sure it works. Do so by adding this link from the sim deployed root: 
  ```
  /wrappers/login/?wrapper=record&validationRule=validateDigits&&numberOfDigits=5&promptText=ENTER_A_5_DIGIT_NUMBER
  ```   
  Further instructions in QA Book
- Please at least load all wrappers except studio in the wrapper suite with internet explorer. 
- make sure to do a complete metacog recording/playback test. Ask @zepumph if you need help.

<h3>Link(s)</h3>

- **[Wrapper Index](https://phet-dev.colorado.edu/html/gravity-force-lab/2.2.0-rc.1/phet-io/)**
- **[Test Matrix](https://docs.google.com/spreadsheets/d/15Galf54GoI--GuRreo_p-hzIsZ7MlitRmiwc4dIUctg/edit#gid=1404448337)**

<hr>

</details>

<details>
<summary><b>Accessibility (a11y) RC Test</b></summary>

<h3>What to Test</h3>

- Specific instructions can be found below.
- Make sure the a11y feature doesn't negatively affect the sim in any way.
- Test all possible forms of input.
  - Test all mouse/trackpad inputs.
  - Test all touchscreen inputs.
  - Test all keyboard navigation inputs (if applicable).
  - Test all forms of input with a screen reader (if applicable).

<h3>Issues to Verify</h3>

- [ ] [Might be an issue with the radio button group](https://github.com/phetsims/gravity-force-lab/issues/223)

These issues should have the "status:ready-for-qa" label. Check these issues off and close them if they are fixed. Otherwise, post a comment in the issue saying that it wasn't fixed and link back to this issue. If the label is "status:ready-for-review" or "status:fixed-pending-deploy" then assign back to the developer when done, even if fixed.

<h3>Link(s)</h3>

- **[a11y View](https://phet-dev.colorado.edu/html/gravity-force-lab/2.2.0-rc.1/phet/gravity-force-lab_a11y_view.html)**
- **[Simulation](https://phet-dev.colorado.edu/html/gravity-force-lab/2.2.0-rc.1/phet/gravity-force-lab_all_phet.html)**
- **[Test Matrix](https://docs.google.com/spreadsheets/d/1dPp2zsj4WW577vQNgCnx9d3cPNxh_JrJj_EQ8CaNqOc/edit#gid=1975614346)**

<h3>Screen Readers</h3>

This sim supports screen readers. If you are unfamiliar with screen readers, please ask Katie to introduce you to screen readers. If you simply need a refresher on screen readers, please consult the [QA Book](link), which should have all of the information you need as well as a link to a screen reader tutorial made by Jesse. Otherwise, look over the a11y view before opening the simulation. Once you've done that, open the simulation and make sure alerts and descriptions work as intended.

<h3>Platforms and Screen Readers to Be Tested</h3>

- [x] Windows 10 + Latest Chrome + Latest JAWS
- [x] Windows 10 + Latest Firefox + Latest NVDA
- [x] macOS + Safari + VoiceOver
- [x] iOS + Safari + VoiceOver (Please test this! Especially with the Ruler interaction)

<h3>Critical Screen Reader Information</h3>

We are tracking known screen reader bugs in [this Google Document](https://drive.google.com/open?id=12kTs-g-iKEIH1dyG7Q41_W_oNL4gUKbkW-IQgZjMUBw). If you find a screen reader bug, please check it against this list.

<h3>Keyboard Navigation</h3>

This sim supports keyboard navigation. Please make sure it works as intended on all platforms by itself and with a screen reader.

<h3>Final Requests</h3>

- [x] If this sim is in [this list](https://github.com/phetsims/perennial/blob/master/data/accessibility), make an issue in [website](https://github.com/phetsims/website) to ask if PhET research page links need updating. Please assign to @terracoda and @emily-phet.

<hr>

</details>

<details>
<summary><b>FAQs for QA Members</b></summary>

<br>
  
  

  <details>
  <summary><i>There are multiple tests in this issue... Which test should I do first?</i></summary> 
  
  Test in order! Test the first thing first, the second thing second, and so on.

  </details>

  <br>

  

  <details>
  <summary><i>How should I format my issue?</i></summary>
  
  Here's a template for making issues:

      <b>Test Device</b>

      blah

      <b>Operating System</b>

      blah

      <b>Browser</b>

      blah

      <b>Problem Description</b>

      blah

      <b>Steps to Reproduce</b>

      blah

      <b>Visuals</b>

      blah

      <details>
      <summary><b>Troubleshooting Information</b></summary>

      blah

      </details>

  </details>

  <br>

  

  <details>
  <summary><i>Who should I assign?</i></summary>
  
  We typically assign the developer who opened the issue in the QA repository.

  </details>
  
  <br>

  

  <details>
  <summary><i>My question isn't in here... What should I do?</i></summary>
  
  You should:

  1. Consult the [QA Book](https://github.com/phetsims/QA/blob/master/doc/qa-book.md).
  2. Google it.
  3. Ask Katie.
  4. Ask a developer.
  5. Google it again.
  6. Cry.

  </details>

<br>

<hr>

</details>

**Comment by ariel-phet** [comment]
Concurrent top priority with #469

**Comment by KatieWoe** [comment]
IE performance is rather poor when dragging the masses or the sliders around. Not really worth an issue in my opinion. @zepumph let me know if you disagree and want one made.

**Comment by KatieWoe** [comment]
With mobile VO, if enhanced sound is on, it can be hard to tell while the sliders are moving since three sounds are playing at once.

**Comment by zepumph** [comment]
>With mobile VO, if enhanced sound is on, it can be hard to tell while the sliders are moving since three sounds are playing at once.

Can you describe what the three sounds are? I only hear the mass "boomp" and the force "waaaah"

**Comment by emily-phet** [comment]
Since it's with VO on, maybe she's referring to the description as the third sound?

**Comment by KatieWoe** [comment]
Yes. So description, boomp, and waaaah, with boomp being hard to detect in the mix.

**Comment by terracoda** [comment]
@KatieWoe by boomp, do you mean the mass sound that is added when Enhanced Sound is checked? If so, that's interesting because, I think that sound has been more salient for blind users. It's been the force sound that is a little less obvious to detect.

I haven't looked too carefully at all the sound interview data for this sim, but if I am recalling correctly, several interviewees noticed that there was more than one sound happening when mass was being changed.

**Comment by KatieWoe** [comment]
I was quoting @zepumph above. He was using it for the mass sound under enhanced sound.

**Comment by terracoda** [comment]
And I haven't seen any interviews using mobile VO with this sim.

**Comment by KatieWoe** [comment]
Memory Test results:
![memresults](https://user-images.githubusercontent.com/41024075/73476199-0a269700-434f-11ea-9a99-70c4cbaaa66f.png)

**Comment by KatieWoe** [comment]
I looked at NVDA on the simulation wrapper in phet-io quickly. Didn't see any immediate problems. @zepumph let me know if you want a thorough testing screen readers and phet-io. I did wonder how they would handle changes made in studio, but I can't really check due to https://github.com/phetsims/studio/issues/92

**Comment by KatieWoe** [comment]
Studio doesn't work with IE. I believe this is expected.

**Comment by zepumph** [comment]
That is expected. Will you add that to the qa book.

**Comment by lmulhall-phet** [comment]
On Firefox, the sound is quieter for the first few seconds. Let me know if you'd like an issue made @zepumph.

**Comment by lmulhall-phet** [comment]
Sorry for the confusion as to who was doing macOS 10.12 + Safari. It's done now. QA is done with the testing matrices for the general RC and PhET-iO, but @KatieWoe might want further testing before we call an end.

**Comment by KatieWoe** [comment]
QA is done

**Comment by zepumph** [comment]
Awesome!!! Thank you!

**Comment by chrisklus** [comment]
https://github.com/phetsims/chipper/commit/348ef4d3679e1e02e4b7ee918ffb738a7ca22573 needs to be cherry picked for the next RC. Once that's complete and `studio` is added to the list of dependencies for this RC, the `studio` sha should be changed from master to the sha of the repo at the time that the rest of the RC shas were taken.

**Comment by zepumph** [comment]
Alright. Everything here is all set, I'm going to make rc.2 now.

**Comment by zepumph** [comment]
Thanks QA!
